src/Symfony/Component/Cache/Adapter/CouchbaseBucketAdapter.php

```diff
@@ -0,0 +1,226 @@
+    private const MAX_KEY_LENGTH = 250;
+    private const KEY_NOT_FOUND = 13;
+
+    /** @var Bucket */
+    private $bucket;
+
+    /** @var MarshallerInterface */
+    private $marshaller;
+
+    public function __construct(\CouchbaseCluster $client, string $bucket, string $namespace = '', int $defaultLifetime = 0, array $options = [])
+    {
+        $this->init($client, $bucket, $namespace, $defaultLifetime, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function createConnection($servers, array $options = []): \CouchbaseCluster
+    {
+        if (\is_string($servers)) {
+            $servers = [$servers];
+        } elseif (!\is_array($servers)) {
+            throw new InvalidArgumentException(sprintf('CouchbaseAdapter::createClient() expects array or string as first argument, %s given.', \gettype($servers)));
+        }
+
+        if (!static::isSupported()) {
+            throw new CacheException('Couchbase >= 2.6.0 is required');
+        }
+
+        set_error_handler(function ($type, $msg, $file, $line) { throw new \ErrorException($msg, 0, $type, $file, $line); });
+
+        $newServers = [];
+        try {
+            $username = $options['username'] ?? '';
+            $password = $options['password'] ?? '';
+
+            foreach ($servers as $dsn) {
+                if (\is_array($dsn)) {
+                    continue;
+                }
+
+                if (0 !== strpos($dsn, 'couchbase:')) {
+                    throw new InvalidArgumentException(sprintf('Invalid Couchbase DSN: %s does not start with "couchbase:"', $dsn));
+                }
+
+                preg_match('/^(?<protocol>couchbase(s)?):\/\/((?<username>[^@:]*+):(?<password>[^@]*+)@)?(?<cluster>.+)(\/.*)?/m', $dsn, $matchers);
+
+                $username = $matchers['username'] ?: $username;
+                $password = $matchers['password'] ?: $password;
+                $newServers += explode(',', $matchers['cluster']);
+            }
+
+            $servers = $newServers ?? $servers;
+            $connectionString = $matchers['protocol'].'://'.implode(',', $servers);
+
+            $client = new \CouchbaseCluster($connectionString);
+            $client->authenticateAs($username, $password);
+
+            return $client;
+        } finally {
+            restore_error_handler();
+        }
+    }
+
+    /**
+     * @return bool
+     */
+    public static function isSupported(): bool
+    {
+        return \extension_loaded('couchbase') && version_compare(phpversion('couchbase'), '2.6.0', '>=');
+    }
+
+    /**
+     * @param \CouchbaseCluster        $client
+     * @param string                   $bucket
+     * @param string                   $namespace
+     * @param int                      $defaultLifetime
+     * @param array                    $options
+     * @param MarshallerInterface|null $marshaller
+     *
+     * @throws CacheException
+     */
+    private function init(
```

Comment:
no need for the method, let's do it in the constructor

Reply:
ok, thank you.